utils/sl/components/__init__.py

```diff
@@ -299,9 +298,9 @@ def define_components(reqs):
     retriever = GitRepoRetriever("https://github.com/spdk/spdk.git", True)
     reqs.define('spdk',
                 retriever=retriever,
-                commands=['cd dpdk; '                            \
-                          'git fetch; '                          \
-                          'git checkout origin/spdk-19.11.6',    \
+                # git -C would be a better option here, however does not play
+                # with older versions of git.
+                commands=['git --git-dir dpdk/.git --work-tree dpdk checkout origin/spdk-19.11.6', \
```

Comment:
Per the previous comment, I think the previous syntax was doing it's job.  IMHO, it's much more clear about what is going on than using obscure `git` command line options to try to achieve the same.  My preference would be to revert this for transparency and readability without having to refer to a manpage.

Reply:
I slightly prefer this syntax as it uses less sh to do it's job, but it would be nice if we could use the -C option.  Your other analysis was correct however and I can achieve what I was looking for with no changes to this file, so I've backed the whole thing out, including the extra 'git clean' in argobots.